client/landing/gutenboarding/onboarding-block/design-selector/index.tsx

```diff
@@ -31,37 +28,93 @@ export default () => {
+			} else {
+				nextLayouts.add( layout.slug );
+			}
+			return nextLayouts;
+		} );
+
+	const resetState = () => {
+		setSelectedDesign( undefined );
+		resetLayouts();
+	};
+
+	const transitionTiming = 250;
+	const hasSelectedDesign = !! selectedDesign;
+	const [ cp, setCp ] = useState< number >();
 
-	const [ selectedDesign, setSelectedDesign ] = useState< string | undefined >();
 	return (
-		<div className="design-selector">
-			<h1 className="design-selector__title">
-				{ NO__( 'Choose a starting design for your site' ) }
-			</h1>
-			<h2 className="design-selector__subtitle">
-				{ NO__( "You'll be able to customize your new site in hundreds of ways." ) }
-			</h2>
-			<div className="design-selector__grid">
-				{ homepageTemplates.map( template => (
-					<Card
-						className={ classnames( 'design-selector__design-option', {
-							'is-selected': template.slug === selectedDesign,
-						} ) }
-						key={ template.slug }
-						isElevated
-						onClick={ () => setSelectedDesign( template.slug ) }
-					>
-						<CardMedia>
-							<img
-								alt={ template.title }
-								src={ template.preview }
-								srcSet={ srcSet( template.preview, [ gridWidth / 2, gridWidth / 4 ] ) }
-							/>
-						</CardMedia>
-					</Card>
-				) ) }
+		<div className={ classnames( 'design-selector', { 'has-selected-design': selectedDesign } ) }>
+			<div
+				className="design-selector__header-container"
+				onClick={ () => resetState() }
+				aria-hidden={ hasSelectedDesign ? 'true' : undefined }
+			>
+				<h1 className="design-selector__title">
+					{ NO__( 'Choose a starting design for your site' ) }
+				</h1>
+				<h2 className="design-selector__subtitle">
+					{ NO__( "You'll be able to customize your new site in hundreds of ways." ) }
+				</h2>
 			</div>
+
+			<CSSTransition in={ ! hasSelectedDesign } timeout={ transitionTiming }>
+				<div
+					className="design-selector__grid-container"
+					onClick={ hasSelectedDesign ? resetState : undefined }
+				>
+					<div className="design-selector__grid">
+						{ designs.map( design => (
+							<DeisgnCard
+								key={ design.slug }
+								design={ design }
+								isSelected={ design.slug === selectedDesign?.slug }
+								style={
+									selectedDesign?.slug === design.slug
+										? {
+												transform: `translate3d( 0, calc( -100vh + ${ -( cp ?? 0 ) + 10 }px ), 0 )`,
+										  }
+										: undefined
+								}
+								onClick={ ( e: MouseEvent< HTMLDivElement > ) => {
+									window.scrollTo( 0, 0 );
+									setCp( e.currentTarget.offsetTop );
+									setSelectedDesign( currentTemplate =>
+										currentTemplate?.slug === design?.slug ? undefined : design
+									);
+									resetLayouts();
+								} }
+							/>
+						) ) }
+					</div>
+				</div>
+			</CSSTransition>
+
+			<CSSTransition in={ hasSelectedDesign } timeout={ transitionTiming } unmountOnExit>
+				<div className="page-layout-selector__container">
+					<PageLayoutSelector
```

Comment:
As a follow up we should think if makes sense to display the PageLayoutSelector as a [Popover](https://developer.wordpress.org/block-editor/components/popover/) and use `onFocusOutside` callback to reset design selection state.